marian-nmt/marian-dev
# Add hand vectorised avx512 binary operators

### Description
This pull requests adds avx512 vectorisation for binary operators

List of changes:
- AVX512 vectorisations
- Small optimisations for avx2 and sse cases

Added dependencies: none

### How to test
Please run this on one of your servers with avx512 support to see if you actually get some performance benefits from it

### Checklist


I have attached the output of the failed regression tests: [regtest.tar.gz](https://github.com/marian-nmt/marian-dev/files/9283033/regtest.tar.gz)
